keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -307,13 +307,15 @@ class SqlDeliveryConfigRepository(
             )
             .onDuplicateKeyIgnore()
             .execute()
-          // delete any other environment's link to this same resource (in case we are moving it
-          // from one environment to another)
+          // delete any other environment's link to any version of this same resource (in case we
+          // are moving it from one environment to another)
           jooq.deleteFrom(ENVIRONMENT_RESOURCE)
             .where(ENVIRONMENT_RESOURCE.ENVIRONMENT_UID.notEqual(environmentUID))
             .and(
-              ENVIRONMENT_RESOURCE.RESOURCE_UID.equal(
-                select(RESOURCE.UID).from(RESOURCE).where(RESOURCE.ID.eq(resource.id))
+              ENVIRONMENT_RESOURCE.RESOURCE_UID.`in`(
+                select(RESOURCE.UID)
+                  .from(RESOURCE)
+                  .where(RESOURCE.ID.eq(resource.id))
```

Comment:
I could have added `.and(RESOURCE.VERSION.eq(resource.version))` to this sub-query but that version may have changed, so I think it makes more sense to remove _all_ references regardless of the version.